cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -288,9 +337,10 @@ class device_row_comparator {
  */
 template <typename Comparator, weak_ordering... values>
 struct weak_ordering_comparator_impl {
-  __device__ bool operator()(size_type const lhs, size_type const rhs) const noexcept
+  template <typename... Ts>
+  __device__ constexpr bool operator()(Ts const... args) const noexcept
   {
-    weak_ordering const result = comparator(lhs, rhs);
+    weak_ordering const result = comparator(args...);
```

Comment:
I'm confused. Why is this variadic now? Can this callable take anything other than 2 arguments?

Reply:
This weak ordering layer can accept two weakly or strongly typed indices. That is, `size_type, size_type` or `lhs_index_type, rhs_index_type` or `rhs_index_type, lhs_index_type`. It might be better to express that with two template parameters and some SFINAE requirements on the types.

Reply:
Implemented non-variadic template parameters with SFINAE requirements in 75249e876714c069fed6ab7ba08f52a760768743.

Reply:
tbh, I think the SFINAE is a bit unnecessary and will likely just inflate compile time. I think it's fine to just leave it as two unconstrained template types. If the types don't match the expectations of `comp` it will fail to compile anyways.

Reply:
Okay! Fixed in bed11626d408e3f7907dd97f7451b01881066614.